### ISSUE TITLE
dtreeviz fails with IndexError for n_classes >= 11

### ISSUE THREAD

**Issue by CermakM** [parrt/dtreeviz#26]
Apparently `dtreeviz` in `dtreeviz.trees.py` fails on the following block of code:

```python
> /home/macermak/.local/lib/python3.6/site-packages/dtreeviz/trees.py(320)dtreeviz()
    318 
    319     n_classes = shadow_tree.nclasses()
--> 320     color_values = color_blind_friendly_colors[n_classes]
    321 
    322     # Fix the mapping from target value to color for entire tree
```

resulting in

```python
IndexError: list index out of range
```

My model has __`15` classes__ and I believe the length of `color_blind_friendly_colors` (which is __`11`__) is causing the issue.

**Comment by parrt** [comment]
ah. yeah, sorry. I should put a check in there but only 10 classes can be handled right now.  we had to handpick the colors.  sorry about that.

**Comment by CermakM** [comment]
Wow, quick response!

So there is nothing I can do at the moment?

**Comment by parrt** [comment]
Nope :( sorry.

**Comment by CermakM** [comment]
Okay, let's hope for a quick solution in the bright future. Keep up the good work and thanks for the quick response.

**Comment by parrt** [comment]
thanks :)  I'd be happy to accept a pull request that increased the number of color palettes ;)

**Comment by Ashafix** [comment]
It's a bit more tricky than just having more colors.

`get_num_bins` gets called with `n_classes` as well and it references `NUM_BINS` which also only has 10 values.
What would be a good default value for `NUM_BINS` for `n_classes` > 10?

Any good ideas about having more colors? 
Just repeating the colors and giving a warning, or making up new colors?

**Comment by parrt** [comment]
Yeah, That's the problem. I started running out of ideas for colors and default values above 10.  May be a sequential color map is required.

**Comment by EricCacciavillani** [comment]
Currently running into the same issue for an idea I am testing. 
But totally understandable. Maybe a semi-simple and quick workaround is just to declare a simple target to color hex dict. 

`{0: #fcba03,
   1: #fc03a9}`

**Comment by chenhajaj** [comment]
Any solution? I even considered not using colors, but when I add None to the color_blind_friendly_colors array (colors.py), I get the following error:
 color_map = {v: color_values[i] for i, v in enumerate(class_values)}

TypeError: 'NoneType' object is not subscriptable

**Comment by parrt** [comment]
In principle, just add another list of colors so this works:

```
color_values = color_blind_friendly_colors[n_classes]
```

**Comment by chenhajaj** [comment]
> In principle, just add another list of colors so this works:
> 
> ```
> color_values = color_blind_friendly_colors[n_classes]
> ```

I have 30 classes. I don't really care about colors, they can all be represented by the same color.
Can I make it work?

**Comment by parrt** [comment]
See if you can set `dtreeviz.color_blind_friendly_colors[30+1] = [c]*30` for some color `c`.

**Comment by chenhajaj** [comment]
> See if you can set `dtreeviz.color_blind_friendly_colors[30+1] = [c]*30` for some color `c`.

Worked, thanks :)
Had to change the NUM_BINS array as well

**Comment by benelot** [comment]
Maybe allow to represent only certainty of the written class via the hue. Then you can either have hue="class_certainty" or hue="certainty", and it switches to hue="certainty" above 10 classes and probably drops a warning that it did.

**Comment by tlapusan** [comment]
Hi folks, if you still need a solution for this issue, you can follow this PR https://github.com/parrt/dtreeviz/pull/185

**Comment by 0ptimista** [comment]
@tlapusan tla, @parrt 

Hi, I really appreciate the work  making support  on more classes. 

I now have a 86 classes decision tree model. Is it any possible way to call `decision_boundaries` without color issue?( "Index Error")

Or It's a bad idea having 86 classes ,   should I make my classes smaller?
